src/coreclr/gc/objecthandle.cpp

```diff
@@ -439,29 +442,27 @@ void CALLBACK ScanPointerForProfilerAndETW(_UNCHECKED_OBJECTREF *pObjRef, uintpt
         rootFlags |= kEtwGCRootFlagsPinning;
         break;
 
+#ifdef FEATURE_VARIABLE_HANDLES
     case    HNDTYPE_VARIABLE:
-#ifdef FEATURE_NATIVEAOT
-    {
-        // Set the appropriate ETW flags for the current strength of this variable handle
-        uint32_t nVarHandleType = GetVariableHandleType(handle);
-        if (((nVarHandleType & VHT_WEAK_SHORT) != 0) ||
-            ((nVarHandleType & VHT_WEAK_LONG) != 0))
-        {
-            rootFlags |= kEtwGCRootFlagsWeakRef;
-        }
-        if ((nVarHandleType & VHT_PINNED) != 0)
         {
-            rootFlags |= kEtwGCRootFlagsPinning;
-        }
+            // Set the appropriate ETW flags for the current strength of this variable handle
+            uint32_t nVarHandleType = GetVariableHandleType(handle);
+            if (((nVarHandleType & VHT_WEAK_SHORT) != 0) ||
+                ((nVarHandleType & VHT_WEAK_LONG) != 0))
+            {
+                rootFlags |= kEtwGCRootFlagsWeakRef;
+            }
+            if ((nVarHandleType & VHT_PINNED) != 0)
+            {
+                rootFlags |= kEtwGCRootFlagsPinning;
+            }
 
-        // No special ETW flag for strong handles (VHT_STRONG)
-    }
-#else
-        _ASSERTE(!"Variable handle encountered");
```

Comment:
assume this assert is no longer necessary?

Reply:
It can be moved to the `default:` case.